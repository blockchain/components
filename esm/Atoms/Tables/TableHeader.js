function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-direction: row;\n  justify-content: flex-start;\n  align-items: center;\n  width: 100%;\n  padding: 10px 30px;\n  box-sizing: border-box;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
var BaseTableHeader = styled.div(_templateObject());

var TableHeader = function TableHeader(_ref) {
  var children = _ref.children;
  return React.createElement(BaseTableHeader, null, children);
};

TableHeader.propTypes = {
  children: PropTypes.node
};
export default TableHeader;