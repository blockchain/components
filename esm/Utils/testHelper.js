import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}