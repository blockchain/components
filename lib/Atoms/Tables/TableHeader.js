"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-direction: row;\n  justify-content: flex-start;\n  align-items: center;\n  width: 100%;\n  padding: 10px 30px;\n  box-sizing: border-box;\n  color: ", ";\n  text-transform: uppercase;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var BaseTableHeader = _styledComponents.default.div(_templateObject(), function (props) {
  return props.theme['tableHeaderColor'];
});

var TableHeader = function TableHeader(_ref) {
  var children = _ref.children;
  return _react.default.createElement(BaseTableHeader, null, children);
};

TableHeader.propTypes = {
  children: _propTypes.default.node
};
var _default = TableHeader;
exports.default = _default;