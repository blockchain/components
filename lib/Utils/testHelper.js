"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}