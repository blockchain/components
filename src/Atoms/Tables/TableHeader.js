import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  padding: 10px 30px;
  box-sizing: border-box;
  color: ${(props) => props.theme['tableHeaderColor']};
  text-transform: uppercase;
`

const TableHeader = ({ children }) => <Wrapper>{children}</Wrapper>

TableHeader.propTypes = {
  children: PropTypes.node,
}

export default TableHeader
