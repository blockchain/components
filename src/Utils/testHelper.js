import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
