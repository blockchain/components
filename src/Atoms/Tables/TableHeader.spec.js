import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import TableHeader from './TableHeader'

describe('TableHeader', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<TableHeader>MyTableHeader</TableHeader>)

    expect(tree).toMatchSnapshot()
  })
})
