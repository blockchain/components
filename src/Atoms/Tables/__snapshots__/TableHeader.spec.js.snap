// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 10px 30px;
  box-sizing: border-box;
  color: #999B9E;
  text-transform: uppercase;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentShadowColor": "#545456",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <TableHeader>
    <styled.div>
      <div
        className="c0"
      >
        MyTableHeader
      </div>
    </styled.div>
  </TableHeader>
</ThemeProvider>
`;
